connector/apmconnector/metric_to_metric_connector.go

```diff
@@ -0,0 +1,257 @@
+// Copyright New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package apmconnector // import "github.com/newrelic/opentelemetry-collector-components/connector/apmconnector"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type OpenTelemetryMetricToApmMetricConnector struct {
+	config *Config
+	logger *zap.Logger
+
+	metricsConsumer consumer.Metrics
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) ConsumeMetrics(ctx context.Context, td pmetric.Metrics) error {
+	metrics := ConvertMetrics(c.logger, c.config, td)
+	return c.metricsConsumer.ConsumeMetrics(ctx, metrics)
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Start(_ context.Context, _ component.Host) error {
+	c.logger.Info("Starting the APM Metric Connector")
+	if c.config.ApdexT == 0 {
+		c.config.ApdexT = defaultApdexT
+	}
+	return nil
+}
+
+func (c *OpenTelemetryMetricToApmMetricConnector) Shutdown(context.Context) error {
+	c.logger.Info("Stopping the APM Metric Connector")
+	return nil
+}
+
+func ConvertMetrics(logger *zap.Logger, config *Config, md pmetric.Metrics) pmetric.Metrics {
```

Comment:
since we're adding new metrics, we should update `metadata.yaml`

Reply:
These actually aren't new metrics. They're the same ones generated before but instead of from spans they're generated from other metrics. That said, looks like the metadata.yaml is missing many of the metrics that have been added previously.

Is metadata.yaml just for generating the code and tests under the metadata directory? It doesn't look like we're using any of that code at the moment. Mind if we defer updating metadata.yaml until we're closer to settling on what will be generated by the collector vs. what we will generate on the backend?